source/slang/slang-doc-markdown-writer.cpp

```diff
@@ -223,17 +223,39 @@ String DocMarkdownWriter::_getFullName(Decl* decl)
 
 String _translateNameToPath(const UnownedStringSlice& name)
 {
+    // We will generate an all-lowercase file name based on the name of the decl.
+    // We do so by converting all capital letters to lowercase, and append a disambiguator
+    // postfix at the end that tracks the location of the capital letters in the original name.
+    // For example, `MyType` will be converted to `mytype-02`, because there is one capital letter
+    // at location 0 and one at location 2.
+    // To prevent URL issues, all leading underscores are replaced with '0', and we
+    // will also append the location of the replaced character to the disambiguator postfix.
+    // For example, `_MyType` will be converted to `0mytype-013`.
+    // To keep disambiguators short, we use base 36 to represent each location.
+
```

Comment:
When the name is longer than 10 characters, the post fix numbers may become ambiguous. "_012" could mean 0, 1, 2 or 0, 12.
Maybe we should add an underscore character for each, like _0_1_2.

Reply:
we use base 36, so it will be until 36 characters when potential ambiguity can arise. I am intentionally trading potential ambiguity with conciseness here.

Reply:
Given that 36 characters is pretty long for most identifiers we care about, I think this is good enough.
Here is how long 36 characters is: `abcdefghijklmnopqrstuvwxyz0123456`.